pkg/util/mon/bytes_usage.go

```diff
@@ -495,16 +500,18 @@ func (mm *BytesMonitor) Start(ctx context.Context, pool *BytesMonitor, reserved
 
 	var effectiveLimit int64
 	if pool != nil {
-		// If we have a "parent" monitor, then register mm as its child by
-		// making it the head of the doubly-linked list.
-		pool.mu.Lock()
-		if s := pool.mu.head; s != nil {
-			s.parentMu.prevSibling = mm
-			mm.parentMu.nextSibling = s
+		if enableMonitorTreeTracking {
+			// If we have a "parent" monitor, then register mm as its child by
+			// making it the head of the doubly-linked list.
+			pool.mu.Lock()
+			if s := pool.mu.head; s != nil {
+				s.parentMu.prevSibling = mm
+				mm.parentMu.nextSibling = s
+			}
+			pool.mu.head = mm
+			pool.mu.numChildren++
```

Comment:
We're only incrementing `numChildren` if `enableMonitorTreeTracking`, but we decrement it in `doStop` unconditionally. Will that cause problems?

Reply:
This is only used to size `children := make([]*BytesMonitor, 0, mm.mu.numChildren)`, so it's not a big deal.

Reply:
That's a good catch. I don't think it can cause any problems (because this is only needed when populating `crdb_internal.node_memory_monitors` table via `TraverseTree` call, and when the env var is disabled, we don't link children of `root` monitor to the root, and IIUC those children are long-living (i.e. never stopped, or on the node shutdown), so `numChildren` should never become negative for `root`). This omission was introduced because of a very recent refactor on master. I think it'd be nice to include that change into backports. Thanks for the catch.